cmd/postgres_exporter/queries.go

```diff
@@ -119,13 +119,15 @@ var queryOverrides = map[string][]OverrideQuery{
 			`
 			SELECT slot_name, database, active, pg_xlog_location_diff(pg_current_xlog_location(), restart_lsn)
 			FROM pg_replication_slots
+                        WHERE pg_is_in_recovery() = False
```

Comment:
Hi Nurian,

This change is for 2 cases:
- v16+. 
In this case as you said we can have that v16 check and run without pg_is_in_recovery() condition

- any version when using pg_failover_slots (extension from EDB)
In this case we need the pg_is_in_recovery() condition for any versions (this extension will copy the replication slot to the replica, making that pre-v16 version behave like a v16)

If I do the change you are requesting it would not solve the issue for the second case, while keeping the condition as proposed would cover both.

Reply:
Got it, thank you